experimental/segmenter/src/word.rs

```diff
@@ -41,11 +41,28 @@ pub struct WordBreakIterator<'l, 's, Y: RuleBreakType<'l, 's> + ?Sized>(
 
 derive_usize_iterator_with_type!(WordBreakIterator);
 
+/// The word type derived from the rule status in a rule-based word segmenter.
```

Comment:
Suggestion: this is public doc, so describing an implementation detail is probably not very useful. How about something like:  ``/// The word type tag that is returned by [`WordBreakIterator::word_type()`].``